mne/io/eyelink/_utils.py

```diff
@@ -5,6 +5,252 @@
+
+    Parameters
+    ----------
+    token : str
+        Single element from tokens list.
+    """
+    if isinstance(token, str):
+        try:
+            float(token)
+            return True
+        except ValueError:
+            return False
+    else:
+        raise ValueError(
+            "input should be a string," f" but {token} is of type {type(token)}"
+        )
+
+
+def _convert_types(tokens):
+    """Convert the type of each token in list.
+
+    The tokens input is a list of string elements.
+    Posix timestamp strings can be integers, eye gaze position and
+    pupil size can be floats. flags token ("...") remains as string.
+    Missing eye/head-target data (indicated by '.' or 'MISSING_DATA')
+    are replaced by np.nan.
+
+    Parameters
+    ----------
+    Tokens : list
+        List of string elements.
+
+    Returns
+    -------
+        Tokens list with elements of various types.
+    """
+    return [
+        int(token)
+        if token.isdigit()  # execute this before _isfloat()
+        else float(token)
+        if _isfloat(token)
+        else np.nan
+        if token in (".", "MISSING_DATA")
+        else token  # remains as string
+        for token in tokens
+    ]
+
+
+def _parse_line(line):
+    """Parse tab delminited string from eyelink ASCII file.
+
+    Takes a tab deliminited string from eyelink file,
+    splits it into a list of tokens, and converts the type
+    for each token in the list.
+    """
+    if len(line):
+        tokens = line.split()
+        return _convert_types(tokens)
+    else:
+        raise ValueError("line is empty, nothing to parse")
+
+
+def _is_sys_msg(line):
+    """Flag lines from eyelink ASCII file that contain a known system message.
+
+    Some lines in eyelink files are system outputs usually
+    only meant for Eyelinks DataViewer application to read.
+    These shouldn't need to be parsed.
+
+    Parameters
+    ----------
+    line : string
+        single line from Eyelink asc file
+
+    Returns
+    -------
+    bool :
+        True if any of the following strings that are
+        known to indicate a system message are in the line
+
+    Notes
+    -----
+    Examples of eyelink system messages:
+    - ;Sess:22Aug22;Tria:1;Tri2:False;ESNT:182BFE4C2F4;
+    - ;NTPT:182BFE55C96;SMSG:__NTP_CLOCK_SYNC__;DIFF:-1;
+    - !V APLAYSTART 0 1 library/audio
+    - !MODE RECORD CR 500 2 1 R
+    """
+    return any(["!V" in line, "!MODE" in line, ";" in line])
+
+
+def _get_sfreq(rec_info):
+    """Get sampling frequency from Eyelink ASCII file.
+
+    Parameters
+    ----------
+    rec_info : list
+        the first list in self._event_lines['SAMPLES'].
+        The sfreq occurs after RATE: i.e. [..., RATE, 1000, ...].
+
+    Returns
+    -------
+    sfreq : int | float
+    """
+    for i, token in enumerate(rec_info):
+        if token == "RATE":
+            # sfreq is the first token after RATE
+            return rec_info[i + 1]
```

Comment:
I would be tempted to do just
```suggestion
    return rec_info[rec_info.index("RATE") + 1]
```
if you really want to return `None` when `RATE` is not present, try/except `ValueError` that gets raised by `.index` when it's not in the list

Reply:
I didn't know about the `index` method. yep sounds good!